crates/fluvio-cluster/src/runtime/local/k8s.rs

```diff
@@ -0,0 +1,211 @@
+use std::{
+    fs::File,
+    path::{PathBuf, Path},
+    process::{Command, Stdio},
+};
+
+use fluvio_command::CommandExt;
+use k8_config::{K8Config, User, UserDetail, Cluster, ClusterDetail, Context, ContextDetail};
+use rcgen::{Certificate, CertificateParams, IsCa, DistinguishedName, DnType, DnValue, KeyUsagePurpose};
+use tracing::{info, debug};
+
+use super::LocalRuntimeError;
+
+const DEFAULT_K8S_CONTEXT_NAME: &str = "fluvio-local";
+const DEFAULT_K8S_CLUSTER_NAME: &str = "fluvio-local-cluster";
+const DEFAULT_K8S_USER_NAME: &str = "fluvio-local-client";
+const DEFAULT_K8S_NAMESPACE: &str = "default";
+
+pub struct K8sProcess {
+    pub log_dir: PathBuf,
+    pub data_dir: PathBuf,
+    pub base: PathBuf,
+    pub etcd_endpoint_url: String,
+}
+
+impl K8sProcess {
+    pub fn start(&self) -> Result<(String, Certs), LocalRuntimeError> {
+        let logs_path = self.log_dir.join("flv_k8s.log");
+        let outputs = File::create(&logs_path)?;
+        let errors = outputs.try_clone()?;
+
+        let port = portpicker::pick_unused_port()
```

Comment:
This seems to be repeated.  can we just make it common util?

Reply:
Done